src/pages/admin/users/EditUser.tsx

```diff
@@ -0,0 +1,80 @@
+import { useState } from "react";
+import { useQuery } from "react-query";
+import { useNavigate, useParams } from "react-router";
+import Button from "../../../components/Button";
+import Layout from "../../../components/Layout";
+import UserForm from "../../../components/UserForm";
+import useAxios from "../../../lib/hooks/useAxios";
+import { useAdmin } from "../../../lib/hooks/useRole";
+
+const EditUser = () => {
+  useAdmin();
```

Comment:
Please rename to be more self-explanatory

```suggestion
  useRequireAdmin(); // or useRequireAdminRole()
```